File: 0001-0500/0228_summaryRanges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        first = nums[0]
        last = nums[0]
        i = 1
        answer = []
        while i <= len(nums):
            if i < len(nums) and nums[i] == nums[i-1] + 1:
                last = nums[i]
            else:
                if first == last:
                    answer.append(str(first))
                else:
                    answer.append(str(first) + "->" + str(last))
                if i < len(nums):
                    first = nums[i]
                    last = nums[i]
            i += 1
        return answer

File: 0001-0500/0228_summaryRanges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> result = new ArrayList<>();
        if (nums.length == 0) {
            return result;
        }
        String temp = Integer.toString(nums[0]);
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i-1] + 1) {
                if (temp.length() < 3 || ! (temp.length() >= 3 && temp.substring(temp.length()-2, temp.length()).equals("->"))) {
                    temp += "->";
                }
            } else {
                if (temp.length() >= 3 && temp.substring(temp.length()-2, temp.length()).equals("->")) {
                    temp += Integer.toString(nums[i-1]);
                }
                result.add(temp);
                temp = Integer.toString(nums[i]);
            }
        }
        if (temp.length() >= 3 && temp.substring(temp.length()-2, temp.length()).equals("->"))  {
            temp += Integer.toString(nums[nums.length-1]);
        }
        result.add(temp);
        return result;
    }
}
